### ISSUE TITLE
More operations for `GraphInfo`

### ISSUE THREAD

**Issue by yebai** [TuringLang/AbstractPPL.jl#53]
In #47, we implemented a few basic operations for `GraphInfo`, for example, construction, `getindex`. For practical purposes, we need to be able to 

- `sample` from `GraphInfo` model, i.e. sample all stochastic nodes from their prior distribution
- `condition/decondition`, i.e. fix values of certain nodes to user-specified value
-  `logdensityof`, i.e. compute the log joint probability of a `GraphInfo` model given values of all nodes. 

A good testing case for these features would be 

1. define a model
2. initialise the model by sampling from the prior
3. compute the log density of the sampled parameters

**Comment by yebai** [comment]
@PavanChaggar this would be a nice follow-up PR for https://github.com/TuringLang/AbstractPPL.jl/pull/47, but supporting sampling depends on #51 I think.

**Comment by phipsgabler** [comment]
Transferred to SymbolicPPL.jl
